# HG changeset patch
# User Gerald Squelart <gsquelart@mozilla.com>
# Date 1451894747 -3600
#      Mon Jan 04 09:05:47 2016 +0100
# Node ID 7e497a0b15cdfeca00bd1b9ad234785899073a4c
# Parent  c2133ad4a73860ef18662c5144777e76652d4cde
Bug 1224371 - Cast uint8_t to uint32_t before shift. r=jya

Note: C-style cast because it is C code.

diff --git a/media/libvpx/vp9/decoder/vp9_decoder.c b/media/libvpx/vp9/decoder/vp9_decoder.c
--- vp9/decoder/vp9_decoder.c
+++ vp9/decoder/vp9_decoder.c
@@ -494,16 +494,16 @@ vpx_codec_err_t vp9_parse_superframe_ind
         decrypt_cb(decrypt_state, x, clear_buffer, frames * mag);
         x = clear_buffer;
       }
 
       for (i = 0; i < frames; ++i) {
         uint32_t this_sz = 0;
 
         for (j = 0; j < mag; ++j)
-          this_sz |= (*x++) << (j * 8);
+          this_sz |= (uint32_t)(*x++) << (j * 8);
         sizes[i] = this_sz;
       }
       *count = frames;
     }
   }
   return VPX_CODEC_OK;
 }
